### PR TITLE
version-info: make run under python3

### LINKED ISSUE #1507: Tools: Python 3ize version_info.py
* Seen in version (release version or githash): 
* On flight controller (if applicable): 

Issue details:

### PR DESCRIPTION
**Pull request by mlyle**
Fixes #1507

Functional change: sacrificed lowercasing in the author sort operation, as that'd be a pain to deal with in unicode across 2 & 3.

Also save a couple of passes of running the versioninfo (still is run once per flight target, though, which is a whole heck of a lot of duplicated work scanning git stuff).

### CODE REVIEW

File: make/scripts/version-info.py [hunk 1]
```diff
@@ -139,7 +148,7 @@ def _get_authors_since_release(self):
 
             self._exec('shortlog -s {}..HEAD'.format(last_rel['ref']))
             if self._rc == 0:
-                self._authors_since_release = sorted([l.strip().split(None, 1)[1] for l in self._out.splitlines()], key=str.lower)
+                self._authors_since_release = sorted([l.strip().split(None, 1)[1] for l in self._out.splitlines()])
```
Comment (tracernz) [inline make/scripts/version-info.py:151]: You have explicitly decoded `self._out` to string in `_exec` so the string sort should should be operating on Python `str` type in both 2 & 3? Tested on 2.7.12 and 3.5.2 and both worked fine with the sort key.

**Comment by tracernz** [comment]
Nice changeset. :+1:  I like the multiple input files although the help strings could do with some clarification. ;)

File: make/scripts/version-info.py (hunk 1 above)
Comment (mlyle) [reply make/scripts/version-info.py:151]: Not sure as to the difference in your environment but on Python2 I get:

```
  File "/home/mlyle/dRonin/make/scripts/version-info.py", line 151, in _get_authors_since_release
    self._authors_since_release = sorted([l.strip().split(None, 1)[1] for l in self._out.splitlines()], key=str.lower)
TypeError: descriptor 'lower' requires a 'str' object but received a 'unicode'
```

Comment (tracernz) [reply make/scripts/version-info.py:151]: Okay, safer to leave that then. :)

**Review by tracernz — APPROVED** [review]

**Pull request merged**